### DISCUSSION TITLE
Move to organization

### DISCUSSION THREAD

**Discussion by jcbhmr** [Andrew-Chen-Wang/github-wiki-action#19] [General]
<details><summary>outdated</summary>

my own musings on this in https://github.com/jcbhmr/publish-wiki-action-OLD/issues/8

I would be interesting in merging some of the features (namely the retainment of commits, history, messages, etc) from my feeble attempt at a wiki github action into this repo https://github.com/jcbhmr/publish-wiki-action-OLD

I don't know if @spenserblack is also interested in consolidating the ecosystem, but this repo seems to be:
1. the most popular github wiki -related action out there (162 dependents https://github.com/Andrew-Chen-Wang/github-wiki-action/network/dependents) and
2. actively monitored by the maintainer 👍❤

Here's a table of other possibles that I had considered:

> Here's a comparison table that I've made:
> 
> ![image](https://user-images.githubusercontent.com/61068799/223895708-8584edcf-0282-4da1-bd7e-682570d90eb3.png)
> 
> * https://github.com/Andrew-Chen-Wang/github-wiki-action is the most popular
> * https://github.com/spenserblack/actions-wiki is the most up-to-date
> * https://github.com/newrelic/wiki-sync-action has its own target audience (with bidi sync feature)
> * https://github.com/kai-tub/external-repo-sync-action kinda feels like it's _too_ generalized

Due to it being the de-facto github wiki publish action, i'd be interested in contributing and possibly reaching out to others (like @spenserblack) to possibly consolidate the ecosystem. I think this would be a good idea because:

1. it means that there can be _multiple contributors_ (like the people you bring on as collaborators from other wiki actions) who can
    - inject new ideas into the repo
    - help address any issues/prs in a more round-robin way instead of bottlenecking on one person who may or may not happen o go on vacation for a week or something
2. mean that we can more robustly claim that this is the defacto wiki publishing action

</details>

**Comment by jcbhmr** [comment]
As @Andrew-Chen-Wang said in https://github.com/Andrew-Chen-Wang/github-wiki-action/pull/9

> Yea I agree, splitting out would be nice, possibly with a github org instead of personal repo.

If you're open to that organization idea, we **might** be able too get in touch with @wow-actions & @bubkoo (I think they are the primary maintainer) https://github.com/wow-actions/.github/issues/1 and put it under that umbrella? GitHub does an excellent job of 301 redirects for moved repos, so I don't think the transfer would be the problem; the problem would be getting maintenance, ownership and other governance/code-style stuff figured out. This would be a relatively **small** undertaking since the 301 redirects do all the heavy lifting. The hard part is communication with @wow-actions, @bubkoo, @Andrew-Chen-Wang and anyone else involved.

But that's just an idea. Maybe sit on the idea for a few weeks. No rush! Let things cool down a little until you have the bandwidth for such a change.

**Comment by jcbhmr** [comment]
I'm reviving this topic once again @Andrew-Chen-Wang because I wanna try again lol 😁. I think that organizations like @JazzBand @unjs or @unifiedjs etc are a great way to take the burden off of one maintainer and spread it over many people. I think that this would be a good idea, specifically because @jcbhmr seems to be the main contributor despite this being under @Andrew-Chen-Wang.

I think that this could also kill two birds with one stone if this package were transferred and renamed at the same time to just "some-actions-org/publish-wiki" so that #65 could be solved with a sister "some-actions-org/pull-wiki" action! Win-win! It solves the feature creep problem (via clear name) and gets this repo more attention (from users and from contributors)!

Here's some ideas of organizations:
- https://github.com/actions-cool
- https://github.com/actions-ecosystem
- https://github.com/actionshub
- https://github.com/wow-actions

**Reply by spenserblack** [reply]
@jcbhmr You might want to keep `@`ing organizations to a minimum. Pretty sure you're generating notifications for users that aren't related to this discussion.

But are you talking about moving repos to an existing org, or creating a new one?

**Reply by Andrew-Chen-Wang** [reply]
I'm a pretty active contributor to Jazzband, and these orgs aren't really that helpful. Creating an org is beneficial if you have multiple repos. I think for now, this is fine.

**Reply by jcbhmr** [reply]
> these orgs aren't really that helpful.

I think they can be very helpful for letting someone else take over _WITHOUT_ needing to Transfer a repo or bug the original author about stuff that you can't edit. It makes it easier to slide the "who's maintaining this project" around.

> Creating an org is beneficial if you have multiple repos.

👆 This actually something that I want to press more. With just the act of "push to github wiki", that's pretty atomic. With the addition of "rewrite links AND push to github wiki" that should really be two separate actions because there are _other wiki services like GitLab wiki_ that might still want to rewrite links for but _not push to_. Or hey, maybe you don't want to rewrite links. Having them as separate actions makes sense.

With the potential addition of #66 I think that's going even more overboard into one action that does a whopping _four things_.

I think that this is a great time when these massive scope increases are being considered to potentially make the migration to an organization to make it so that:
1. There's 4 actions instead of 1
2. Thus the actions _can be composed with other stuff_ instead of one lump-sum action.
3. All the actions have a "backing org" if @jcbhmr ever looses interest or Andrew-Chen-Wang ever goes on vacation or whatever there's another admin to move things along

I'd even be open to keeping this action as Andrew-Chen-Wang/github-wiki-action but then using something like this to put the code elsewhere. This solves my problems of 1) splitting into parts (so that I can use specifically the wiki pull WITHOUT link transforms or whatever) and 2) the organization ease-of-collaboration issue.

```yml
runs:
  using: composite
  steps:
    - if: inputs.mode == 'push'
      uses: actions-org/prepare-links-for-wiki@v1
    - if: inputs.mode == 'push'
      uses: actions-org/push-wiki@v1
    - if: inputs.mode == 'pull'
      uses: actions-org/pull-wiki@v1
    - if: inputs.mode == 'pull'
      uses: actions-org/prepare-links-for-github-source@v1
```

---

@spenserblack I don't think @nodefill 👈 organizations get pings? I think it's just teams like @nodefill/fake-team that can get pings. I honestly don't know. But if you're right, I should stop doing useless pings, yes.

As for whether a new org or existing org, I was _hoping_ to integrate with an existing organization since there's already some well-established "community actions"-like organizations that I think this action would be great for.

**Reply by xorima** [reply]
as the owner of actions hub, I got the ping, and if you want it over in my org and to join to help maintain it I'm all good. If you want to reach out to me https://keybase.io/xorima

**Comment by jcbhmr** [comment]
So far I've split out the complicated source->wiki and wiki->source preprocessing logic into two individual actions:
https://github.com/jcbhmr/preprocess-github-wiki-for-source
https://github.com/jcbhmr/preprocess-source-for-github-wiki
